neo/Ledger/MemoryPool.cs

```diff
@@ -4,162 +4,358 @@
+                    _unverifiedSortedHighPriorityTransactions.Remove(item);                        
+                }
+
+                if (DateTime.UtcNow > reverifyCutOffTimeStamp) break;
             }
+
+            return addedCount;
         }
 
-        public bool TryGetValue(UInt256 hash, out Transaction tx)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private int ReverifyLowPriorityTransactions(int count, double secondsTimeout, Snapshot snapshot)
         {
-            if (_mem_pool_free.TryGetValue(hash, out PoolItem item))
+            DateTime reverifyCutOffTimeStamp = DateTime.UtcNow.AddSeconds(secondsTimeout);
+            int addedCount = 0;            
+            
+            foreach (PoolItem item in _unverifiedSortedLowPriorityTransactions.Reverse().Take(count).ToArray())
             {
-                tx = item.Transaction;
-                return true;
+                // Re-verify the top fee max low priority transactions that can be verified in a block
+                if (item.Transaction.Verify(snapshot, _unsortedTransactions.Select(p => p.Value.Transaction)))
+                {
+                    _unsortedTransactions.TryAdd(item.Transaction.Hash, item);
+                    _sortedLowPrioTransactions.Add(item);
+                    _unverifiedTransactions.TryRemove(item.Transaction.Hash, out _);
+                    _unverifiedSortedLowPriorityTransactions.Remove(item);
+                }
+                
+                if (DateTime.UtcNow > reverifyCutOffTimeStamp) break;
             }
-            else if (_mem_pool_fee.TryGetValue(hash, out item))
+
+            return addedCount;
+        }
+
+        /// <summary>
+        /// Reverify up to a given maximum count of transactions. Verifies less at a time once the max that can be
+        /// persisted per block has been reached. 
+        /// </summary>
+        /// <param name="maxToVerify">Max transactions to reverify, the value passed should be >=2. If 1 is passed it
+        ///                           will still potentially use 2.</param>
+        /// <param name="snapshot">The snapshot to use for verifying.</param>
+        /// <returns>true if more unsorted messages exist, otherwise false</returns>
+        public bool ReVerifyTopUnverifiedTransactionsIfNeeded(int maxToVerify, Snapshot snapshot)
+        {
+            if (_unverifiedSortedHighPriorityTransactions.Count > 0)
             {
-                tx = item.Transaction;
-                return true;
+                // Always leave at least 1 tx for low priority tx
+                int verifyCount = _sortedHighPrioTransactions.Count > MaxHighPriorityTxsPerBlock || maxToVerify == 1
+                    ? 1 : maxToVerify - 1; 
+                maxToVerify -= ReverifyHighPriorityTransactions(verifyCount, MaxSecondsToReverifyHighPrioTxPerIdle, 
+                    snapshot);
+                
+                if (maxToVerify == 0) maxToVerify++;
```

Comment:
Is this for ensuring  the verification of, at least, 1 Low Priority tx?

Reply:
Yes.